include/fixed_containers/struct_view.hpp

```diff
@@ -0,0 +1,1045 @@
+inline constexpr std::size_t MAX_PATH_LENGTH = 16;
+inline constexpr std::size_t MAX_DIM = 5;
+inline constexpr std::size_t MAX_CONTAINER_SIZE = 2000;
+inline constexpr std::string_view ITERABLE_PATH_NAME = "data[:]";
+inline constexpr std::string_view OPTIONAL_PATH_NAME = "value()";
+inline constexpr std::string_view PATH_DELIMITER = ".";
+
+// users may want to have a unified way to refer to a name to some member functions
+inline constexpr std::string_view SIZE = "size()";
+inline constexpr std::string_view HAS_VALUE = "has_value()";
+
+// string_view for metadata type, we use std::string_view over enum class for extensibility
+using StructTreeNodeType = std::string_view;
+// string_view for detail type for type matching
+using StructTreeNodeTypeDetail = std::string_view;
+// call interface for metadata type, std::any for extensibility
+using StructTreeNodeCallInterface = std::any;
+
+inline constexpr std::string_view OPTIONAL = "OPTIONAL";
+inline constexpr std::string_view DEFAULT = "DEFAULT";
+inline constexpr std::string_view PRIMITIVE = "PRIMITIVE";
+inline constexpr std::string_view PRIMITIVE_ENUM = "ENUM_PRIMITIVE";
+inline constexpr std::string_view PRIMITIVE_STRING_VIEW = "STRING_VIEW_PRIMITIVE";
+inline constexpr std::string_view ITERABLE = "ITERABLE";
+inline constexpr std::string_view ITERABLE_RESIZABLE = "RESIZABLE_ITERABLE";
+
+using PathNameChain = FixedVector<std::string_view, MAX_PATH_LENGTH>;
+
+template <std::size_t MAXIMUM_SIZE>
+using Indices = FixedVector<std::size_t, MAXIMUM_SIZE>;
+
+template <std::size_t MAXIMUM_SIZE>
+using Strides = FixedVector<std::size_t, MAXIMUM_SIZE>;
+
+template <std::size_t MAXIMUM_SIZE>
+using Shape = FixedVector<std::size_t, MAXIMUM_SIZE>;
+
+template <std::size_t MAXIMUM_SIZE>
+using Capacity = FixedVector<std::size_t, MAXIMUM_SIZE>;
+
+template <std::size_t MAXIMUM_SIZE>
+struct FixedTensorView
+{
+    using Indices = Indices<MAXIMUM_SIZE>;
+
+    std::size_t base_offset{};
+    std::size_t dim{};
+    Strides<MAXIMUM_SIZE> strides{};
+    Capacity<MAXIMUM_SIZE> capacity{};
+
+    [[nodiscard]] auto get_offset(Indices indices) const
+    {
+        return std::inner_product(
+            std::begin(indices), std::end(indices), std::begin(strides), base_offset);
+    }
+
+    constexpr bool operator==(const FixedTensorView&) const = default;
+};
+
+// Adaptors for common private types used by the `struct_view`
+// Now we have supports for `std::optional` and `FixedVector` (ResizableIterator)
+template <typename T>
+class OptionalView
+{
+private:
+    bool has_value_;
+    T value_;
+
+public:
+    using value_type = T;
+    void set_has_value(bool new_has_value) { has_value_ = new_has_value; }
+    bool has_value() const { return has_value_; }
+    const T& value() const { return value_; }
+    T& value() { return value_; }
+
+    template <typename... Args>
+    T& emplace(Args&&... args)
+    {
+        if (has_value_)
+        {
+            value_.~T();
```

Comment:
Nit: `std::destroy_at()`

Reply:
already remove the need of `OptionalView` completely with `emplace_default` and `reset`.